src/components/CandidateForm.jsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react'
+import styles from './CandidateForm.module.css'
+
+const CandidateForm = (props) => {
+    let defaultState = Object.assign({}, ...props.inputData.map(input => {
```

Comment:
This is not necessary here, you are better off to just store everything that comes from the props in state. You will anyway need all the values later, in the formInputs() method. 

So  const [state, setState] = useState(props.inputData) should do it.